tests/errors/semantic/blocking/INTERFACE_WRONG_TYPES.py

```diff
@@ -0,0 +1,15 @@
+# pylint: disable=missing-function-docstring, missing-module-docstring/
+from pyccel.decorators import types
+from pyccel.decorators import template
+
+@template('z', types=['int', 'real'])
+@template('y', types=['int', 'real'])
+@types('z', 'z', 'y')
+def multi_tmplt_1(x, y, z):
```

Comment:
Could you add a similar test where the types of `y` and `z` are not the same.

E.g.
```
@template('y', types=['int'])
@template('z', types=['int', 'real'])
@types('y', 'z')
def multi_tmplt_2(y, z):
    return y + z

multi_tmplt_2(1.5, 3.5)
```

Reply:
By similar I mean also checking we get an error. I just want to make sure that the code correctly restricts the types of y

Reply:
oh, yes now I see. so I should add a test in errors

Reply:
Yes please :)

Reply:
can you check please

Reply:
Looks good thanks. You can resolve this conversation now

Reply:
Ok great